velox/core/SimpleFunctionMetadata.h

```diff
@@ -376,13 +408,65 @@ class SimpleFunctionMetadata : public ISimpleFunctionMetadata {
   }
 
  private:
+  struct SignatureTypesAnalysisResults {
+    std::vector<std::string> argsTypes;
+    std::string outputType;
+    std::set<std::string> variables;
+    TypeAnalysisResults::Stats stats;
+  };
+
+  SignatureTypesAnalysisResults analyzeSignatureTypes() {
+    std::vector<std::string> argsTypes;
+
+    TypeAnalysisResults results;
```

Comment:
minor nit (feel free to ignore):  It seems like TypeAnalysisResults exists only so that you can pass a single argument to TypeAnalysis.run, rather than the Stats, an std::set<string> for variables maintained in this function, and an ostringstream local to the lambda below.  

I think (not a strong opinion) that it might be cleaner to break them up, so you don't have to call resetTypeString() in the lambda (that seems error prone).